Src/BootCamp.Chapter/Credentials.cs

```diff
@@ -1,22 +1,54 @@
-﻿namespace BootCamp.Chapter
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace BootCamp.Chapter
 {
     // TODO: make a struct and add validation and other needed methods (if needed)
-    public class Credentials
+    public readonly struct Credentials
     {
-        public string Username;
-        public string Password;
+        public readonly string Username;
+        public readonly string Password;
 
         public Credentials(string username, string password)
         {
+            if (string.IsNullOrEmpty(username) || string.IsNullOrEmpty(password)) throw new ArgumentException("user name and password not be empty.");
```

Comment:
It's one hell of a line 😄 
Try to be more specific at what is wrong, don't make people guess. An improved version of this would be:

```cs
if(string.IsNullOrEmpty(username) throw new ArgumentException(...);
if(string.IsNullOrEmpty(password) throw new ArgumentException(...);
...
```

Reply:
Haha you are right I'll be more carefull about it.